### PR TITLE
Ruby 2.4 compatibility

### PR DESCRIPTION
**Pull request by execjosh**
- Use `Integer` instead of `Fixnum`
- Add ruby 2.4.0 to Travis tests

### CODE REVIEW

File: lib/flexirest/request.rb [hunk 1]
```diff
@@ -224,7 +224,7 @@ def call(explicit_parameters=nil)
 
     def prepare_params
       params = @params || @object._attributes rescue {}
-      if params.is_a?(String) || params.is_a?(Fixnum)
+      if params.is_a?(String) || params.is_a?(Integer)
```
Comment (andyjeffries) [inline lib/flexirest/request.rb]: Does this now only work for Ruby 2.4, or is this change compatible with 1.9.3 and upwards as per the rest of Flexirest?

**Comment by andyjeffries** [comment]
Thanks for the work, I've commented on one of the commits, regarding compatibility with versions before 2.4.

File: lib/flexirest/request.rb (hunk 1 above)
Comment (execjosh) [inline lib/flexirest/request.rb]: As far as I can tell it should be compatible!  [`Integer`][1] exists in 1.9.3.

Here's a loop I've used to test locally:

```bash
#!/bin/bash

while read -r ver; do
  export RBENV_VERSION="$(basename "$ver")"
  ruby -e 'puts %{#{RUBY_VERSION} -- #{[1234,-1234,12345678901234567890,-12345678901234567890,1.1,0].map{|v| [v.class,v.is_a?(Integer)]}}}'
done < <(find ~/.rbenv/versions -mindepth 1 -maxdepth 1 -type d)
```

Which gives me:

```
1.9.3 -- [[Fixnum, true], [Fixnum, true], [Bignum, true], [Bignum, true], [Float, false], [Fixnum, true]]
2.0.0 -- [[Fixnum, true], [Fixnum, true], [Bignum, true], [Bignum, true], [Float, false], [Fixnum, true]]
2.1.0 -- [[Fixnum, true], [Fixnum, true], [Bignum, true], [Bignum, true], [Float, false], [Fixnum, true]]
2.2.0 -- [[Fixnum, true], [Fixnum, true], [Bignum, true], [Bignum, true], [Float, false], [Fixnum, true]]
2.3.1 -- [[Fixnum, true], [Fixnum, true], [Bignum, true], [Bignum, true], [Float, false], [Fixnum, true]]
2.4.1 -- [[Integer, true], [Integer, true], [Integer, true], [Integer, true], [Float, false], [Integer, true]]
```

[1]: http://ruby-doc.org/core-1.9.3/Integer.html

**Comment by andyjeffries** [comment]
Merged in as the test failures don't seem related to Ruby 2.4 support.

**Pull request merged**